src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -711,6 +711,10 @@ public String toString()
                           .toString();
     }
 
+    // TODO would it make sense to rename? This is used in multiple contexts and incorrectly labeled composite types
+    // as literal, which led to their incorrect encoding in the trie. The main use for this method is to determine
+    // if the type requires a trie or a kd tree. There are also encoding implications, but as seen in this PR, those
+    // weren't used correctly.
```

Comment:
I agree if the purpose of this method is to check whether the index requires a trie or kd tree, it should be named more directly. I'm thinking about something like `indexKind` returning enum `{ KD_TREE, TRIE }`.

Reply:
I wonder if we should defer on changing this in this PR. It seems like this logic could go into the `OnDiskFormat` interface since we could change formatting from one version to another. I see that as a real possibility if someone started storing a bunch of large varint or decimal types since SAI won't do very well for those right now.

Reply:
I am deferring this to another PR.